app/com/gu/viewer/proxy/LiveProxy.scala

```diff
@@ -2,24 +2,60 @@ package com.gu.viewer.proxy
 
 import com.gu.viewer.config.AppConfig
 import com.gu.viewer.logging.Loggable
+import play.api.mvc.Cookie
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
+import java.time.{LocalDateTime, ZoneOffset}
 
-class LiveProxy(proxyClient: ProxyClient, config: AppConfig)(implicit ec: ExecutionContext) extends Loggable {
+class LiveProxy(proxyClient: ProxyClient, config: AppConfig)(implicit
+    ec: ExecutionContext
+) extends Loggable {
 
   val serviceHost = config.liveHost
 
+  private def setCookieListForNonAdvertisingBanner(
```

Comment:
Wonder if there's a way of avoid the repetition here? Perhaps this fn could be added to a `Proxy` trait that both Live and Draft extend?